### PR TITLE
Fix metadata file name

### LINKED ISSUE #49: Upgrade fails with `autoalias` error
### What happened?

Attempting to upgrade `pulumi-newrelic`, I got this error message:

```
𝛌 upgrade-provider pulumi-newrelic
---- Setting Up Environment ----
- ✓ GOWORK="off": done
- ✓ PULUMI_MISSING_DOCS_ERROR="true": done
---- Discovering Repository ----
- Ensure 'github.com/pulumi/pulumi-newrelic'
  - ✓ Expected Location: /Users/ianwahbe/go/src/github.com/pulumi/pulumi-newrelic
  - ✓ Downloading: skipped
  - ✓ Validating: done
- pull default branch
  - ✓ /usr/local/bin/git ls-remote --heads origin: done
  - ✓ finding default branch: master
  - ✓ /usr/local/bin/git checkout master: done
  - ✓ /usr/local/bin/git pull origin: done
- ✓ Repo kind: patched
- ✓ Planning Provider Update: 3.20.0 -> 3.20.2
- ✓ Planning Bridge Update: v3.43.0 -> v3.44.3
error: unable implement migration 'autoalias': open /Users/ianwahbe/go/src/github.com/pulumi/pulumi-newrelic/provider/cmd/pulumi-resource-pulumi-newrelic/bridge-metadata.json: no such file or directory
```

I'm calling special attention to the path to the file it checked: `pulumi-resource-pulumi-newrelic`. This should be `pulumi-resource-newrelic`.

### Expected Behavior

Upgrade works.

### Steps to reproduce

Attempt to run the `upgrade-provider` tool on `pulumi-newrelic` v3.20.0`.

### Output of `pulumi about`

NA. `upgrade-provider` is at `2b4e08ac8d01517bb4ebbfbde8870bcf13beb104`.

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by aq17**
Fixes #49

### CODE REVIEW

File: upgrade/steps.go [hunk 1]
```diff
@@ -836,6 +836,7 @@ func migrationSteps(ctx Context, repo ProviderRepo, providerName string, descrip
 
 func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
 	steps := []step.Step{}
+	providerName = strings.TrimPrefix(providerName, "pulumi-")
```
Comment (aq17) [inline upgrade/steps.go:839]: Removed in the previous refactor by accident

Comment (iwahbe) [reply upgrade/steps.go:839]: This is also going to change line L847. Is that correct?

Also, we shouldn't do anything while producing a `step.Step`. We need to put L839-845 into a `step.F`.

Comment (aq17) [reply upgrade/steps.go:839]: Yes

File: upgrade/steps.go [hunk 2]
```diff
@@ -837,12 +837,16 @@ func migrationSteps(ctx Context, repo ProviderRepo, providerName string, descrip
 func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
 	steps := []step.Step{}
 	metadataPath := fmt.Sprintf("%s/cmd/pulumi-resource-%s/bridge-metadata.json", *repo.providerDir(), providerName)
-	if _, err := os.Stat(metadataPath); os.IsNotExist(err) {
-		_, err = os.Create(metadataPath)
-		if err != nil {
-			return nil, err
+	steps = append(steps, step.F("initializing bridge-metadata.json", func() (string, error) {
```
Comment (iwahbe) [inline upgrade/steps.go:840]: We could declare the `initializing bridge-metadata.json` step and the `git add metadataPath` step as part of the `steps` literal.

```go
initializeBridgeMetadata := step.F(...)
addBridgeMetadata := step.Cmd(...)
steps := []step.Step{initializeBridgeMetadata, addBridgeMetadata}
```

We could also declare them inline.

```go
steps := []step.Step{ 
	step.F(...),
	step.Cmd(...),
}
```

Lets do one or the other here.

**Review by iwahbe — APPROVED** [review]

**Pull request merged**